workspaces/ui-v2/src/store/selectors/documentationEditSelectors.ts

```diff
@@ -0,0 +1,35 @@
+import { RootState } from '../root';
+import { IContribution } from '<src>/types';
+
+// Valid changes dedupes:
+// - same contribution value as already set
```

Comment:
We aren't currently doing this since all the state currently isn't in redux - this is _possible_ but kind of convoluted to get the relevant state / initial values for each contribution.

This is identical behavior to how it's currently implemented, however once we move all state into redux this is easier (not trivial, since contribution values will be stored in lots of different locations based on how they are fetched - e.g. fields, bodies, shapes, etc).